source/PackageCache.js

```diff
@@ -157,7 +157,6 @@ module.exports = {
 	// // Tools
 	_normalizePackageList(packageList, packageVersions = {}) {
 		const normalized = packageList
-			.map(packageName => /[^/]+/.exec(packageName)[0])
```

Comment:
I'm a bit concerned about this change. It looked like this `.map` call was intentionally added, but it only seems to cause problems from my perspective; and I didn't see any tests fail when I removed it. Was the original purpose overcome by events, or am I missing something important?

Reply:
This actually is useful! It's for requiring files other than the package's entry point:

```js
#require uuid@3.3.0
const uuid = npm['uuid/v4'];
cli.tell('New unique identifier: ' + uuid());
```

It's pretty uncommon and apparently getting even more rare, but it'd be good to keep supporting it. What kind of issues did this cause?

Reply:
NPM packages can have slashes embedded in them, like the `@octokit/rest` package in my examples. Truncating the package name causes the wrong name to show up in package.json.

I'll add your snippet as a new test case to support, and try to come up with a way to support both. Do you have ideas or preferences on that? Perhaps an escape character of sorts to indicate that the slash is part of the package name and not a separator for requiring other files.

It may take me a couple days to get back to this.

Reply:
Oh, I didn't catch that!
Clearly, packages name with slashes in them seem more common than packages than require you to access a sub-file, so we should prioritize the former use case when designing how to distinguish between the two.

An escape character sounds like a good idea. Because of how minimalistic Tasklemon's npm module is, it's really hard to add a way to pass parameters/more info when calling it, so getting the file path to fit nicely in the key would be ideal.
(that's a big part of why version pinning is done through a header line! I tried very hard to allow specifying the version inline, at the same place the package is used, but could never get it to look right)

Looking at npm's [package naming rules](https://www.npmjs.com/package/validate-npm-package-name#naming-rules), non-URL-safe characters are disallowed, so we could use these as separators or escape characters.
A colon separator might be a good option to separate package name from file path:
- `const uuid = npm['uuid:v4']`
- `const exampleTree = npm['@example/tree:v4/es5']`

I'll keep thinking about it, maybe we can do better still, but this already looks pretty good to me!